pylint/__init__.py

```diff
@@ -95,4 +95,25 @@ def modify_sys_path() -> None:
         sys.path.pop(1)
 
 
+def _catch_valueerror(unraisable: sys.UnraisableHookArgs) -> None:
+    """Overwrite sys.unraisablehook to catch incorrect ValueError.
+
+    Python 3.12 introduced changes that sometimes cause astroid to emit ValueErrors
+    with 'generator already executing'. Fixed in Python 3.12.3 and 3.13.
+
+    https://github.com/pylint-dev/pylint/issues/9138
+    """
+    if (
```

Comment:
Throw in a `pragma: no cover`? It's tested in the primer, just not in the coverage run.